main.c

```diff
@@ -200,48 +286,331 @@ int main(void)
+    };
 
-    psa_key_policy_set_usage(&policy, PSA_KEY_USAGE_VERIFY, alg);
-    ASSERT_SUCCESS_PSA(psa_set_key_policy(verify_handle, &policy));
+    const uint8_t hash_input2[] = "";
+    /* SHA-256 hash of an empty string */
+    const uint8_t sha256_expected_hash2[] = {
+        0xe3, 0xb0, 0xc4, 0x42, 0x98, 0xfc, 0x1c, 0x14,
+        0x9a, 0xfb, 0xf4, 0xc8, 0x99, 0x6f, 0xb9, 0x24,
+        0x27, 0xae, 0x41, 0xe4, 0x64, 0x9b, 0x93, 0x4c,
+        0xa4, 0x95, 0x99, 0x1b, 0x78, 0x52, 0xb8, 0x55
+    };
 
-    ASSERT_SUCCESS_PSA(psa_import_key(verify_handle, key_type, pubkey,
-                                      pubkey_len));
+    ASSERT_SUCCESS_PSA(atecc608a_hash_sha256(hash_input1,
+                                             sizeof(hash_input1) - 1,
+                                             sha256_expected_hash1,
+                                             sizeof(sha256_expected_hash1)));
 
-    /* Verify that the signature produced by the secure element is valid. */
-    ASSERT_SUCCESS_PSA(psa_asymmetric_verify(verify_handle, alg, hash,
-                                             sizeof(hash), signature,
-                                             signature_length));
+    ASSERT_SUCCESS_PSA(atecc608a_hash_sha256(hash_input2,
+                                             sizeof(hash_input2) - 1,
+                                             sha256_expected_hash2,
+                                             sizeof(sha256_expected_hash2)));
+
+    printf("test_hash_sha256 succesful!\n");
+exit:
+    return status;
+}
+
+psa_status_t run_tests()
+{
+    psa_status_t status;
+
+    printf("Running tests...\n");
+    ASSERT_SUCCESS_PSA(test_hash_sha256());
+
+    /* Verify that the device has a locked config zone before running tests
+     * that use slots. */
+    ASSERT_SUCCESS_PSA(atecc608a_check_zone_locked(LOCK_ZONE_CONFIG));
+
+    ASSERT_SUCCESS_PSA(test_generate_import());
+    ASSERT_SUCCESS_PSA(test_export_import());
+    ASSERT_SUCCESS_PSA(test_sign_verify());
+    ASSERT_SUCCESS_PSA(test_psa_import_verify());
+
+    /* Verify that the device has a locked data zone before running tests
+     * that use clear text read. */
+    ASSERT_SUCCESS_PSA(atecc608a_check_zone_locked(LOCK_ZONE_DATA));
+
+    /* Slot 8 is usually used as a clear write and read certificate
+     * or signature slot, as it is the biggest one (416 bytes of space). */
+    ASSERT_SUCCESS_PSA(test_write_read_slot(8));
 
-    printf("Verification successful!\n");
     exit:
         return status;
 }
+
+void print_device_info()
+{
+    atecc608a_print_serial_number();
+    atecc608a_print_config_zone();
+    atecc608a_print_locked_zones();
+    printf("\nPrivate key slot in use: %lu, public: %lu\n",
+           atecc608a_private_key_slot, atecc608a_public_key_slot);
+}
+
+bool interactive_loop()
+{
+    char command [80];
+    char *arg;
+    size_t len;
+
+    printf(USAGE);
+    scanf ("%79s",command);
```

Comment:
Style: Add space after `,`.

Does this 79 include `\r\n` or only `\n`?

Reply:
79 non-whitespace characters:
`%s` - `Any number of non-whitespace characters, stopping at the first whitespace character found. A terminating null character is automatically added at the end of the stored sequence.`
So ending the string with any combination of `\r`, `\n` will work, and not terminating it will work too, since the input will be cut off at 79 and a null terminator will be added afterwards.